### DISCUSSION TITLE
Welcome to pouchdb Discussions!

### DISCUSSION THREAD

**Discussion by janl** [pouchdb/pouchdb#8452] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ebenezer-melkamu** [comment]
Hello

I am trying to develop an offline-first application for a POS system, and I want to know the best practice to follow for this app. currently, I am trying to do something like this and I have some questions

![Pouchdb architecture drawio (1)](https://user-images.githubusercontent.com/70259638/154416080-ea5345cd-b73d-4323-9e53-ada0b7b7152d.png)

1. The app is going to be a web application and I want it to run offline in a browser, and the PouchDB database will be distributed for every client, is there any security concern here?
2. Since I want the app to work offline (except for accessing external APIs), do you recommend developing this as a PWA 
3. I am trying to build and run a REST API inside the web app, is this a good practice?
4. Can anyone share a git repo of similar offline-first applications, which are built with React, NodeJS, Express, PouchDB, and CouchDB ?

**Reply by karlomac** [reply]
Hello @AbegaM 

From your diagram above, not sure what "Local Backend" means --  usually its  Frontend --> PouchDB --> CouchDB.   I suppose the local backend is for the 3rd Party api.

If you wish to do this as Mobile-first app, best to create it as a PWA.   To further answer your questions :
1 . "security concern for PouchDB distributed for every client". --> You can imagine that PouchDB runs locally on the device.  so any information stored in PouchDB will be stored on the device which should be subsequently pushed to a CouchDB server.   Therefore the device itself should be secured - being a POS, the assumption then it is basically on a desk or a tablet device possibly bolted to the table.  The only security concern would be where to push the data gathered on the local POS to the CouchDB server that is secured.  So far from my experience, the only way it will be un-secured is if anyone would "spoof" your CouchDB server - which would mean having the credentials to accept the pouchDB sync.

2. Working offline is best if there is really no connectivity at all and sync when connection is available.  It would be better to have the transactions posted on PouchDB and do an auto-sync or periodic sync so your data will always be available or delayed only by a transaction (defined as a posted line item or after completion of the sales i.e. receipt generation)

3.  I don't quite see the point of this -- usually having PouchDb is enough to gather the data you need and exchange data with CouchDB servers (i.e. inventory pricing which can be synced at start of day or on demand,  post transactions POS ) 

4. I use  vue js / Nuxt js / PouchDb  --- > CouchDB  and Strapi CMS for auxiliary data.
